include/seqan3/alignment/matrix/detail/score_matrix_single_column.hpp

```diff
@@ -0,0 +1,246 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::score_matrix_single_column.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/std/ranges>
+#include <vector>
+
+#include <seqan3/alignment/matrix/detail/affine_cell_proxy.hpp>
+#include <seqan3/range/views/repeat_n.hpp>
+#include <seqan3/range/views/zip.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Score matrix for the pairwise alignment using only a single column.
+ * \ingroup alignment_matrix
+ * \implements std::ranges::input_range
+ *
+ * \tparam score_t The type of the score.
+ *
+ * \details
+ *
+ * In many cases it is sufficient to store only a single score column to compute the alignment between two sequences.
+ * Since the alignment is computed iteratively column by column the same memory can be reused for the next score.
+ * This score matrix scores a column for the optimal score and the horizontal score and uses a virtual column that
+ * stores only one value of type `score_t` for the vertical column. Hence, this matrix can only be used for a column
```

Comment:
```suggestion
 * This score matrix stores the complete column for both the optimal and horizontal score, but only stores a single 
 * value for the vertical column. Hence, this matrix can only be used for a column
```

I think the sentence is a bit convoluted and hence hard to understand. The concept of a `virtual column` and the type of the value stored are not too important here?

Reply:
still open?

Reply:
Mhmm strange. I thought i added it to the batch 🤷 I do it now.